### ISSUE TITLE
Incorrect escaped data in SQL when using Query and COALESCE()

### ISSUE THREAD

**Issue by irthunder** [yiisoft/yii2#19468]
### What steps will reproduce the problem?
$q = new \yii\db\Query;
$q->select('p.id, COALESCE(p.id, 0)')
->from('some_table_name p');

### What is the expected result?
SELECT \`p\`.\`id\`, COALESCE(\`p\`.\`id\`, **\`0\`)** FROM \`some_table_name\` \`p\`

### What do you get instead?
SELECT \`p\`.\`id\`, COALESCE(\`p\`.\`id\`, **\`0)\`** FROM \`some_table_name\` \`p\`

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.44
| PHP version      |  7.3
| Operating system | Windows 8.1

**Comment by samdark** [comment]
Looks like a bug. Do you have time for a pull request containing a test and a fix?

**Comment by irthunder** [comment]
Looks like it should be:
`$columns = preg_split('/\s*,\s*(?![^\(]*\))/', trim($columns), -1, PREG_SPLIT_NO_EMPTY);`

**Comment by samdark** [comment]
Yes. Likely that is correct regex. Do you have some time for a pull request?

**Comment by alex-code** [comment]
Not sure if there could be any edge cases with that regex, testing it with `'a, COALESCE(IF(1, 2, 3), IF(4, 5, 6))'` you get 3 strings back.
```
Array
(
    [0] => a
    [1] => COALESCE(IF(1, 2, 3)
    [2] => IF(4, 5, 6))
)
```

Isn't the preferred method for complex selects to use an array to provide the columns? e.g.
```php
$q = new \yii\db\Query;
$q->select(['p.id', 'COALESCE(p.id, 0)'])->from('some_table_name p');
```
